pkg/decorate/topk.go

```diff
@@ -0,0 +1,85 @@
+package decorate
+
+import (
+	"container/heap"
+	"regexp"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// TopK is a decorator that removes metrics not in the top K by value
+type TopK struct {
+	K uint
+	N string
+}
+
+// Decorate removes all but the top K metrics for a given metric name
+func (t TopK) Decorate(mfs []*dto.MetricFamily) {
+	var topk []*metricHeap
+	var idx []int
+	for i, fam := range mfs {
+		if match, _ := regexp.MatchString(t.N, fam.GetName()); match {
```

Comment:
should we simplify this and just use strings.HasPrefix()? I haven't benchmarked it but normally that comparison would be much faster than a regex match

Reply:
Good call.

Ran a bm for kicks
`regexp: 2975 ns/op` vs. `prefix: 0.28 ns/op`